# Comparison of SARS-CoV-2 Workflow Features

| Workflow               | Status                                                                                       | Publishing Date | Source                                                              | SARS-CoV-2 Exclusive | Framework / Library | Seq. Technology      | Can Process Shotgun Data | Can Process Amplicon Data | Read Filtering                     | Contamination Removal | Primer Trimming | De Novo Assembly                              | Scaffolding   | Consensus / Reference Assembly     | Variant Calling / Annotation                                                | Lineage Calling                                | Low Coverage Masking    | Indication of Quality | GUI / Report                                                                                                                                                                 | Comments                                                                                               |
| ---------------------- | -------------------------------------------------------------------------------------------- | --------------- | ------------------------------------------------------------------- | -------------------- | ------------------- | -------------------- | ------------------------ | ------------------------- | ---------------------------------- | --------------------- | --------------- | --------------------------------------------- | ------------- | ---------------------------------- | --------------------------------------------------------------------------- | ---------------------------------------------- | ----------------------- | --------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ |
| ARTIC                  | Unpublished                                                                                  |                 | [link](https://github.com/artic-network/artic-ncov2019)             | Yes                  | Bash                | Nanopore             | No                       | Yes                       | Guppyplex                          | \-                    | Custom script   | \-                                            | \-            | bcftools consensus                 | medaka consensus, medaka snp, medaka variant, nanopolish variants, longshot | \-                                             | Custom script           |                       | \-                                                                                                                                                                           | State of the Art                                                                                       |
| C-VIEW                 | Unpublished                                                                                  |                 | [link](https://github.com/ucsd-ccbb/C-VIEW)                         |                      | AWS                 | Illumina             |                          |                           | samtools                           | samtools              | iVar            | \-                                            | \-            | Yes                                |                                                                             | Pangolin                                       |                         |                       |                                                                                                                                                                              |                                                                                                        |
| Genome Detective       | [Published](https://doi.org/10.1093/bioinformatics/bty695)                                   | 16.08.21        | [link](https://www.genomedetective.com/app/typingtool/cov/)         | No                   |                     | Illumina, Nanopore   | Yes                      | No                        | Trimmomatic                        | Yes                   | \-              | SPAdes, metaSPAdes                            | Yes           | Yes                                | \-                                                                          | Yes                                            |                         |                       | Webapp                                                                                                                                                                       |                                                                                                        |
| HAPHPIPE               | [Published](https://www.mdpi.com/1999-4915/12/7/758)                                         | 03.07.20        | [link](https://github.com/gwcbi/haphpipe)                           | No                   | Python              | Illumina             | Yes                      | No                        | Trimmomatic                        | \-                    | \-              | SPAdes                                        | MUMMER 3+     | Bowtie2                            | \-                                                                          | \-                                             | \-                      |                       |                                                                                                                                                                              |                                                                                                        |
| HAVoC                  | [Published](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04294-2) | 17.07.21        | [link](https://bitbucket.org/auto_cov_pipeline/havoc/src/master/)   | Yes                  | Shell               | Illumina             | Yes                      | No                        | fastp, Trimmomatic                 | \-                    | \-              | \-                                            | \-            | bcftools/samtools                  | LoFreq                                                                      | Pangolin                                       | BED-tools               | \-                    | \-                                                                                                                                                                           |                                                                                                        |
| IDseq                  | [Published](https://academic.oup.com/gigascience/article/9/10/giaa111/5918865)               | 15.08.20        | [link](https://czid.org/)                                           | No                   |                     | Illumina             | Yes                      | No                        | Trimmomatic, PriceSEQ, CD-HIT- DUP | +                     | \-              | SPAdes                                        | \-            | \-                                 | \-                                                                          | \-                                             | \-                      | \-                    | Yes                                                                                                                                                                          |                                                                                                        |
| IPD 2.0                | [Published](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04172-x) | 13.05.21        | [link](http://www.actrec.gov.in/pi-webpages/AmitDutt/IPD/IPD.html)  | No                   | Python              | Illumina, Nanopore   | Yes                      | No                        | fastp, PoreChop, NanoFilt          | +                     | \-              | Megahit                                       | \-            | \-                                 | VarScan2, LoFreq, freebayes, Medaka, SnpEff                                 | Euclidean Distance based with data from GISAID | \-                      | \-                    | Yes                                                                                                                                                                          |                                                                                                        |
| LABRADOR               | [Published](https://doi.org/10.3390/v13122541)                                               | 19.11.21        |                                                                     | No                   |                     |                      |                          |                           | Trimmomatic                        | BWA                   | \-              | Megahit                                       | \-            | \-                                 | \-                                                                          | \-                                             | \-                      | \-                    | Yes                                                                                                                                                                          |                                                                                                        |
| LAZYPIPE               | [Published](https://academic.oup.com/ve/article/6/2/veaa091/6017186)                         | 02.12.20        | [link](https://bitbucket.org/plyusnin/lazypipe/src/master/)         | No                   | Perl, R             | Illumina             | Yes                      | No                        | Trimmomatic, fastp                 | +                     | \-              | Megahit, Velvet, SPAdes                       | +             | \-                                 | \-                                                                          |                                                | \-                      | \-                    | Yes                                                                                                                                                                          | Environmental                                                                                          |
| PiGx                   | [Preprint](https://www.medrxiv.org/content/10.1101/2021.11.30.21266952v1)                    | 02.12.21        | [link](https://github.com/BIMSBbioinfo/pigx_sarscov2_ww)            | Yes                  | Snakemake           |                      |                          |                           | PRINSEQ                            | Yes                   |                 |                                               |               |                                    | LoFreq, VEP                                                                 | \-                                             | \-                      | \-                    | [Yes](https://bimsbstatic.mdc-berlin.de/akalin/AAkalin_pathogenomics/sarscov2_ww_reports/211104_pub_version/index.html)                                                      | Environmental                                                                                          |
| RECoVERY               | [Preprint](https://www.biorxiv.org/content/10.1101/2021.01.16.425365v2)                      | 05.12.21        | [link](https://aries.iss.it/)                                       | Yes                  | Galaxy              | Illumina, Nanopore   |                          |                           | Trimmomatic                        | Bowtie2               | \-              | \-                                            | \-            | iVar                               | iVar, SnpEff                                                                | \-                                             | \-                      | \-                    | \-                                                                                                                                                                           |                                                                                                        |
| SIGNAL                 | [Published](https://pubmed.ncbi.nlm.nih.gov/32824272/)                                       | 15.08.20        | [link](https://github.com/jaleezyy/covid-19-signal)                 | Yes                  | Snakemake           | Illumina             |                          | Yes                       | trim-galore                        | Custom script         | iVar            | \-                                            | \-            | iVar                               | iVar, breseq                                                                | \-                                             | \-                      | \-                    | Yes                                                                                                                                                                          |                                                                                                        |
| Snakelines             | [Published](http://ceur-ws.org/Vol-2962/paper15.pdf)                                         | 24.09.21        | [link](https://github.com/jbudis/snakelines)                        | No                  | Snakemake           | Illumina, (Nanopore) | Yes                       | Yes                       | cutadapt                           | Bowtie2               | cutadapt        | \-                                            | \-            | bcftools                           | BCFtools, GATK                                                              | Pangolin                                       | Yes, 3 bp               |                       | Yes                                                                                                                                                                          | SnakeLines is intended for the wide range of genomic analysis. The cited publication describes only a specific pipeline for SARS-CoV-2 Variant Detection. Overall, the SnakeLines offers [much more](https://arxiv.org/abs/2106.13649).                                                                                                       |
| TAR-VIR                | [Published](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-2878-2)  | 04.06.19        | [link](https://github.com/chjiao/TAR-VIR)                           | No                   | C++, Python         | Illumina             | No                       | Yes                       | \-                                 | \-                    | \-              | PEHaplo                                       | \-            | \-                                 | \-                                                                          | \-                                             | \-                      | \-                    | \-                                                                                                                                                                           | assembly with partial or not complete reference genomes, full haplotype reconstruction                 |
| V-pipe                 | [Published](https://doi.org/10.1093/bioinformatics/btab015)                                  | 20.01.21        | [link](https://github.com/cbg-ethz/V-pipe)                          | No                   | Snakemake           | Illumina             | Yes                      | Yes                       | PRINSEQ                            | \-                    | \-              | Vicuna, SAVAGE                                | \-            | samtools, bcftools, ConsensusFixer | ShoRAH 2, LoFreq                                                            | \-                                             | bcftools consensus      | \-                    | Yes                                                                                                                                                                          | for any virus, full haplotype reconstruction                                                           |
| VEGA                   | [Published](https://peerj.com/articles/12129/)                                               | 06.09.21        | [link](https://github.com/pauloluniyi/VGEA)                         | No                   | Snakemake           |                      |                          |                           | fastp                              | BWA                   | \-              | IVA                                           | shiver/SeqKit | \-                                 | \-                                                                          | \-                                             | \-                      | \-                    | MultiQC                                                                                                                                                                      |                                                                                                        |
| VIRify                 | Unpublished                                                                                  |                 | [link](https://github.com/EBI-Metagenomics/emg-viral-pipeline)      | No                   | Nextflow, CWL       |                      | No                       | No                        | TrimGalore                         | TrimGalore            | TrimGalore      | rnaSPAdes, MEGAHIT, metaSPAdes                | \-            | \-                                 | \-                                                                          | \-                                             | \-                      | \-                    |                                                                                                                                                                              |                                                                                                        |
| ViPR                   | Unpublished                                                                                  |                 |                                                                     |                      |                     |                      |                          |                           |                                    |                       |                 |                                               |               |                                    |                                                                             |                                                |                         |                       |                                                                                                                                                                              |                                                                                                        |
| ViReflow               | Unpublished                                                                                  |                 | [link](https://github.com/niemasd/ViReflow)                         | No                   | AWS, Reflow         | Illumina             | Yes                      | Yes                       | fastp, PRINSEQ                     | \-                    | iVar, pTrimmer  | coronaSPAdes, metaviralSPAdes, rnaviralSpades | \-            | bcftools consensus                 | FreeBayes, iVar Variants, LoFreq                                            | Pangolin                                       | Yes                     | \-                    | \-                                                                                                                                                                           |                                                                                                        |
| nf-core/viralrecon     | Unpublished (nf-core is [published](https://www.nature.com/articles/s41587-020-0439-x))      |                 | [link](https://github.com/nf-core/viralrecon)                       | No                   | Nextflow            | Illumina, Nanopore   | Yes                      | Yes                       | fastp, artic guppyplex             | Kraken2               | iVar, Cutadapt            | SPAdes (all relevant modes), Unicycler, minia | ABACAS            | iVar, bcftools, artic minion       | iVar, BCFTools, artic minion, SnpEff, SnpSift                               | Pangolin, nextstrain                           | samtools, custom script | Yes                   | [MultiQC](https://nf-core-awsmegatests.s3-eu-west-1.amazonaws.com/viralrecon/results-97bebf8fe12e0e802d4468e133f3a2277ceb843c/platform_illumina/multiqc/multiqc_report.html) | Flexible, portable and community driven pipeline tested on various infrastructures including the Cloud |
| ViraPipe               | [Published](https://academic.oup.com/bioinformatics/article/34/6/928/4587582)                | 02.11.17        | [link](https://github.com/NGSeq/ViraPipe)                           | No                   | Spark               |                      |                          |                           | \-                                 | \-                    | \-              | Megahit                                       | \-            | \-                                 | \-                                                                          | \-                                             | \-                      | \-                    | \-                                                                                                                                                                           |                                                                                                        |
| VirusLab               | [Published](https://www.mdpi.com/2673-6284/10/4/27/htm)                                      | 06.11.21        | \-                                                                  | No                   |                     | Illumina, Nanopore   |                          |                           |                                    |                       |                 | \-                                            | \-            | ARTIC/Galaxy                       | SnpEff                                                                      | Pangolin                                       | \-                      | \-                    | Webapp                                                                                                                                                                       |                                                                                                        |
| bjorn                  | Unpublished                                                                                  |                 | [link](https://github.com/andersen-lab/bjorn)                       | Yes                  | Snakemake, Bash     |                      | No                       | No                        | \-                                 | \-                    | \-              | \-                                            | \-            | \-                                 | \-                                                                          | \-                                             | \-                      | \-                    | \-                                                                                                                                                                           |                                                                                                        |
| nCoV_minipipe, CovPipe | [Referenced](https://www.preprints.org/manuscript/202005.0376/v1)                            |                 | [link](https://gitlab.com/RKIBioinformaticsPipelines/ncov_minipipe) | No                   | Snakemake           | Illumina, (Nanopore) | Yes                      | Yes                       | fastp                              | Kraken2               | bamClipper      | \-                                            | \-            | bcftools consensus                 | freebayes, bcftools, SNPeff                                                 | Pangolin                                       | Yes                     | Yes                   | Yes                                                                                                                                                                          |                                                                                                        |
| ncov2019-artic-nf      | Unpublished                                                                                  |                 | [link](https://github.com/connor-lab/ncov2019-artic-nf)             | Yes                  | Nextflow            | Illumina, Nanopore   | Yes                      | Yes                       | Nanopolish                         | \-                    | iVar            | \-                                            | \-            | iVar                               | iVar                                                                        | \-                                             | Yes                     | Yes                   | \-                                                                                                                                                                           |                                                                                                        |
| poreCov                | [Published](https://www.frontiersin.org/articles/10.3389/fgene.2021.711437/full)             | 28.07.21        | [link](https://github.com/replikation/poreCov)                      | Yes                  | Nextflow            | Nanopore             | No                       | Yes                       | NanoPlot, pycoQC                   | Yes                   | ARTIC           | \-                                            | \-            | ARTIC                              | nextstrain                                                                  | Pangolin, nextstrain                           | ARTIC(< 20)             | ARTIC                 | [Yes](https://htmlpreview.github.io/?https://github.com/replikation/poreCov/blob/master/data/figures/index.html)                                                             |                                                                                                        |

## Interpretation of Features

- **Workflow**: The name of the workflow.
- **Status**: If the workflow was published or not. If it was published, it should contain a link to the publication.
- **Publishing Date**: If the workflow was published, the date of publication.
- **Source**: Link to the source code.
- **SARS-CoV-2 Exclusive**: If the workflow only processes SARS-CoV-2 samples.
- **Framework / Library**: Which framework or library is used for the implementation of the workflow.
- **Seq. Technology**: Which sequcencing technology/technologies can be processed with the workflow.
- **Can Process Shotgun Data**: If the workflow can process [shotgun data](https://en.wikipedia.org/wiki/Shotgun_sequencing).
- **Can Process Amplicon Data**: If the workflow can process [amplicon data](https://en.wikipedia.org/wiki/Amplicon#Technology).
- **Read Filtering**: Which tool the workflow uses for [read filtering](https://compgenomr.github.io/book/filtering-and-trimming-reads.html).
- **Contamination Removal**: Which tool is used for contamination removal (e.g. for reads originating from a humans genome).
- **Primer Trimming**: Which tool the workflow uses for amplicon primer trimming.
- **De Novo Assembly**: Which tool the workflow uses for _de novo_ assembly.
- **Scaffolding**: Which tool the workflow uses for scaffolding.
- **Consensus / Reference Assembly**: Which tool the workflow uses for consensus assembly.
- **Variant Calling / Annotation**: Which tool the workflow uses for variant calling and/or annotation.
- **Lineage Calling**: Which tool the workflow uses for lineage calling (such as B.1.1.7, Delta, etc).
- **Low Coverage Masking**: Which tool the workflow uses for low covered positions of the reconstructed genome.
- **Indication of Quality**: If the workflow indicates of quality of the assembled genome, such as large share of N's or low reference similarity.
- **GUI / Report**: If the workflow outputs an aggregated report. If possible, a link to the report.
- **Comments**: What makes the workflow special.
